devito/arguments.py

```diff
@@ -209,8 +206,23 @@ def dtype(self):
 
     @cached_property
     def rtargs(self):
-        size = ScalarArgument("%s_size" % self.name, self, max)
-        return [size]
+        size = ScalarArgument(self.size_name, self, max)
+        start = ScalarArgument(self.start_name, self, max, 0)
+        end = ScalarArgument(self.end_name, self, max)
+        return [size, start, end]
+
+    def promote(self, value):
+        if not isinstance(value, tuple):
+            size, start, end = self.rtargs
+            value = (value, start.default_value, value)
```

Comment:
if not a tuple, then it's something else, and we're returning it. What is it supposed to be? can we assert it?

Reply:
This method is promoting a scalar to a tuple so everything that is not a tuple is treated like a scalar.

Reply:
OK. Perhaps make it private with ``_`` ?